lib/wallet/rpc.js

```diff
@@ -443,6 +444,33 @@ class RPC extends RPCBase {
     return addrs;
   }
 
+  async getAddressInfo(args, help) {
+    if (help || args.length !== 1)
+      throw new RPCError(errs.MISC_ERROR, 'getaddressinfo "address"');
+
+    const valid = new Validator(args);
+    const addr = valid.str(0, '');
+
+    const address = parseAddress(addr, this.network);
+    const script = Script.fromAddress(address);
+    const isWitness = address.isProgram();
+
+    const wallet = this.wallet.toJSON();
+    const isMine = await this.wallet.hasAddress(address);
+    const isScript = script.isScripthash() || script.isWitnessScripthash();
+
+    return {
```

Comment:
If you don't want `witness_version` and `witness_program` to be on the object, even in the case that `JSON.stringify` hasn't removed it because it's `undefined`, you could do something such as:

```js
const result = {
  address: address.toString(this.network),
  scriptPubKey: script ? script.toJSON() : undefined,
  ismine: isMine,
  iswatchonly: wallet.watchOnly,
  isscript: isScript,
  iswitness: isWitness
};

if (isWitness) {
  result.witness_version = address.version;
  result.witness_program = address.hash.toString('hex');
}

return result;
```

Reply:
Also behavior in `bitcoind` here looks like `witness_version` and `witness_program` are not included in the result for results where `iswitness: false`, and is included where `iswitness: true`.

Reply:
I think that this is overall more readable, switching to this style